File: convert_stocks.py
import csv
import json
import os

print("📊 Starting CSV to JSON conversion...\n")

stocks = []

# Read CSV file
with open('EQUITY_L.csv', 'r', encoding='utf-8') as csvfile:
    reader = csv.DictReader(csvfile)
    
    count = 0
    for row in reader:
        symbol = row.get('SYMBOL', '').strip()
        if not symbol:
            continue
            
        stock = {
            'symbol': symbol,
            'name': row.get('NAME OF COMPANY', '').strip(),
            'series': row.get(' SERIES', 'EQ').strip(),
            'isin': row.get(' ISIN NUMBER', '').strip(),
            'faceValue': float(row.get(' FACE VALUE', '10').strip() or '10'),
            'exchange': 'NSE'
        }
        
        stocks.append(stock)
        count += 1
        
        if count % 500 == 0:
            print(f"Processed: {count} stocks...")

print(f"\n✅ Processed {len(stocks)} stocks from CSV")

# Create directory if needed
os.makedirs('src/data', exist_ok=True)
print("📁 Created/verified src/data directory")

# Write JSON file
with open('src/data/nse_stocks.json', 'w', encoding='utf-8') as jsonfile:
    json.dump(stocks, jsonfile, indent=2, ensure_ascii=False)

print(f"✅ Successfully created src/data/nse_stocks.json")
print(f"📊 Total stocks: {len(stocks)}")
print(f"\n🎉 Conversion complete!\n")

# Show sample
print("Sample data (first 5 stocks):")
print(json.dumps(stocks[:5], indent=2, ensure_ascii=False))

File: scripts/convert-stocks.js
const fs = require('fs');
const path = require('path');

console.log('📊 Starting CSV to JSON conversion (No dependencies version)...\n');

// Read the CSV file
const csvPath = path.join(__dirname, '..', 'EQUITY_L.csv');
const csvData = fs.readFileSync(csvPath, 'utf-8');

// Split into lines
const lines = csvData.split('\n');
const headers = lines[0].split(',');

console.log(`📄 Found ${lines.length - 1} lines in CSV`);
console.log(`📋 Headers: ${headers.join(', ')}\n`);

const stocks = [];
let count = 0;

// Process each line (skip header)
for (let i = 1; i < lines.length; i++) {
  const line = lines[i].trim();
  if (!line) continue;
  
  const values = line.split(',');
  
  const symbol = values[0] ? values[0].trim() : '';
  const name = values[1] ? values[1].trim() : '';
  const series = values[2] ? values[2].trim() : 'EQ';
  const isin = values[6] ? values[6].trim() : '';
  const faceValue = values[7] ? parseFloat(values[7].trim()) : 10;
  
  if (symbol) {
    stocks.push({
      symbol: symbol,
      name: name,
      series: series,
      isin: isin,
      faceValue: faceValue,
      exchange: 'NSE'
    });
    
    count++;
    
    // Progress indicator
    if (count % 500 === 0) {
      console.log(`Processed: ${count} stocks...`);
    }
  }
}

console.log(`\n✅ Processed ${stocks.length} stocks from CSV`);

// Create data directory if it doesn't exist
const dataDir = path.join(__dirname, '..', 'src', 'data');
if (!fs.existsSync(dataDir)) {
  fs.mkdirSync(dataDir, { recursive: true });
  console.log('📁 Created src/data directory');
}

// Write to JSON file
const jsonPath = path.join(dataDir, 'nse_stocks.json');
fs.writeFileSync(jsonPath, JSON.stringify(stocks, null, 2));

console.log(`✅ Successfully created ${jsonPath}`);
console.log(`📊 Total stocks: ${stocks.length}`);
console.log(`\n🎉 Conversion complete!\n`);

// Show sample data
console.log('Sample data (first 5 stocks):');
console.log(JSON.stringify(stocks.slice(0, 5), null, 2));
